plugins/Previous/src/Previous.ts

```diff
@@ -0,0 +1,45 @@
+import { Plugin, PluginCallbacks, State, self, PluginCallbacksOnSetArgument } from '@hookstate/core';
+
+const PluginID = Symbol('Previous');
+
+type PreviousInstanceType<T = unknown> = () => (
+    PluginCallbacks
+    & {
+        get(): T | undefined
+        previous: T | undefined
+    }
+)
+
+const PreviousInstance: PreviousInstanceType = () => ({
+  previous: undefined,
+  onSet(data: PluginCallbacksOnSetArgument) {
+    this.previous = data.previous
```

Comment:
this is an interesting challenge here.... I guess this works correctly if `data.path === []`, ie. root state is set. Does it save the correct previous state when nested state is set? Like this:

```tsx
const state = useState({ props: 0 })[self].attach(Previous);
state.props.set(1)
Previous(state).get() // I would expected `{ props: 0 }` here
Previous(state.prop).get() // I would expect `0` here
state.set({ props: 2 })
Previous(state).get() // I would expected `{ props: 1 }` here
Previous(state.prop).get() // I would expect `1` here
```